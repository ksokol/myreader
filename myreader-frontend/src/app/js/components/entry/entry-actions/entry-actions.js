import React from 'react'
import PropTypes from 'prop-types'
import {IconButton} from '../../../components'

export const EntryActions = props => [
  <IconButton
    key='more-toggle'
    type={`expand-${props.showMore ? 'less' : 'more'}`}
    onClick={props.onToggleShowMore}/>,
  <IconButton
    key='seen-toggle'
    type={`check${props.seen ? '-circle' : ''}`}
    onClick={props.onToggleSeen}/>
]

EntryActions.propTypes = {
  showMore: PropTypes.bool,
  onToggleShowMore: PropTypes.func.isRequired,
  seen: PropTypes.bool.isRequired,
  onToggleSeen: PropTypes.func.isRequired
}
