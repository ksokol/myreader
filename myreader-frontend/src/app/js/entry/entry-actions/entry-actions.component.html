<react-component
        name="IconButton"
        props="{type: 'expand-more'}"
        ng-click="$ctrl.toggleMore()"
        ng-if="!$ctrl.showMore">
</react-component>

<react-component
        name="IconButton"
        props="{type: 'expand-less'}"
        ng-click="$ctrl.toggleMore()"
        ng-if="$ctrl.showMore">
</react-component>

<react-component
        name="IconButton"
        props="{type: 'check-circle'}"
        ng-click="$ctrl.onCheckClick(false)"
        ng-if="$ctrl.myItem.seen">
</react-component>

<react-component
        name="IconButton"
        props="{type: 'check'}"
        ng-click="$ctrl.onCheckClick(true)"
        ng-if="!$ctrl.myItem.seen">
</react-component>
