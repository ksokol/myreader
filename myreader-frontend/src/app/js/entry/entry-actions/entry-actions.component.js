(function () {
    'use strict';

    function EntryActionsComponent() {
        var ctrl = this;

        ctrl.$onInit = function () {
            ctrl.item = ctrl.myItem;
        };

        ctrl.$onChanges = function (obj) {
            ctrl.item = obj.myItem.currentValue;
        };

        ctrl.toggleMore = function () {
            ctrl.showMore = !ctrl.showMore;
            ctrl.myOnMore({showMore: ctrl.showMore});
        };

        ctrl.onCheckClick = function (value) {
            ctrl.myOnCheck({item: { seen: value}});
        };
    }

    require('angular').module('myreader').component('myEntryActions', {
        template: require('./entry-actions.component.html'),
        controller: EntryActionsComponent,
        bindings: {
            myItem: '<',
            myOnMore: '&',
            myOnCheck: '&'
        }
    });

    module.exports = 'myreader.entry.entry-actions.component';

})();
