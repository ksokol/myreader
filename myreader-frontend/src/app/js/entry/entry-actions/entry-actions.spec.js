import React from 'react'
import {EntryActions} from './entry-actions'
import TestRenderer from 'react-test-renderer'

describe('src/app/js/entry/entry-actions/entry-actions.spec.js', () => {

  let props

  beforeEach(() => {
    props = {
      onToggleShowMore: jest.fn(),
      onToggleSeen: jest.fn(),
      seen: true
    }
  })

  const createInstance = () => TestRenderer.create(<EntryActions {...props} />).root

  it('should render expand-more and check-circle icon buttons', () => {
    const instance = createInstance()

    expect(instance.children[0].props).toContainObject({type: 'expand-more'})
    expect(instance.children[1].props).toContainObject({type: 'check-circle'})
  })

  it('should render expand-less and check icon buttons', () => {
    props.seen = false
    props.showMore = true
    const instance = createInstance()

    expect(instance.children[0].props).toContainObject({type: 'expand-less'})
    expect(instance.children[1].props).toContainObject({type: 'check'})
  })

  it('should trigger onToggleShowMore with expand-more icon button', () => {
    createInstance().findByProps({type: 'expand-more'}).props.onClick()

    expect(props.onToggleShowMore).toHaveBeenCalled()
  })

  it('should trigger onToggleSeen with check-circle icon button', () => {
    createInstance().findByProps({type: 'check-circle'}).props.onClick()

    expect(props.onToggleSeen).toHaveBeenCalledWith()
  })

  it('should trigger onToggleShowMore with expand-less icon button', () => {
    props.showMore = true
    const instance = createInstance()
    instance.findByProps({type: 'expand-less'}).props.onClick()

    expect(props.onToggleShowMore).toHaveBeenCalled()
  })

  it('should trigger onToggleSeen with check icon button', () => {
    props.seen = false
    const instance = createInstance()
    instance.findByProps({type: 'check'}).props.onClick()

    expect(props.onToggleSeen).toHaveBeenCalledWith()
  })
})
