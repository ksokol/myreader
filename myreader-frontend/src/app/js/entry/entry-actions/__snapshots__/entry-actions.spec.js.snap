// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/js/entry/entry-actions/entry-actions.spec.js should render expand-less and check icon buttons 1`] = `
Array [
  <IconButton
    onClick={[MockFunction]}
    type="expand-less"
  />,
  <IconButton
    onClick={[MockFunction]}
    type="check"
  />,
]
`;

exports[`src/app/js/entry/entry-actions/entry-actions.spec.js should render expand-more and check-circle icon buttons 1`] = `
Array [
  <IconButton
    onClick={[MockFunction]}
    type="expand-more"
  />,
  <IconButton
    onClick={[MockFunction]}
    type="check-circle"
  />,
]
`;
